lib/phoenix/controller.ex

```diff
@@ -265,7 +265,7 @@ defmodule Phoenix.Controller do
 
   defp get_json_encoder do
     Application.get_env(:phoenix, :format_encoders)
-    |> Keyword.get(:json, Poison)
+    |> Keyword.get(:json, Jason)
```

Comment:
This is backwards incompatible, I am afraid.

Reply:
Yeah, I was going to ask how we should handle it. We can make it an explicit config, but then we'd still need to ship with the phoenix dep, which is not ideal

Reply:
We cannot simply change to Jason as that is backwards incompatible, since Jason is not fully compatible with Poison.

In my opinion, we should expose this function behind `Phoenix.json_encoder` (it can be @doc false) and use it throughout the codebase but we should still default to poison. In new apps, we generate:

    config :phoenix, :format_encoders, json: Jason

Reply:
Another reason to this clean up is that Ecto can only migrate to Jason by default in 3.0. So I am afraid that making this easily achievable is the best way to go for now.